remove references to branch master

gmrabian opened the pull request:
### Description
<!-- Detailed description of changes and related context -->
The `master` branch of CARTS holds what we call CARTS v2. It uses a different pattern and infrastructure than the rest of our MDCT suite. Two years ago we migrated to `main` (CARTS v3). All of our dev infrastructure and code promoted to prod comes from `main`

`master` has been copied to `skipci-archive-master`

We need to remove any references to master in the code so as to avoid errors and confusion.

Once this is merged, github settings are updated, and we've determined it's safe to do so, we will delete the branch named `master`

### Related ticket(s)
<!-- Link to related ticket(s) or issue(s) -->
<!-- Hint: Type CMDCT-<ticket-number> for autolinking -->
CMDCT-3804

---
### How to test
<!-- Step-by-step instructions on how to test, if necessary -->
See if you can find any more references that need to be updated.

Verify CARTS works as expected https://d13roq1oz0l6c0.cloudfront.net/

### Notes
<!-- Changed dependencies, .env files, configs, etc. -->
<!-- Instructions for local dev, e.g. requires new installs in directories -->
Someone with higher permissions will need to remove branch protections and any other git configuration on master

🤔 can you think of a better archived branch name? `skipci-archive-carts-v2`? Also how long do we anticipate keeping this, and for what purpose? I imagine the drift is already large enough that it may not be helpful to reference the old master branch.

---
### Pre-review checklist
<!-- Complete the following steps before opening for review -->
- [ ] ~I have added [thorough](https://shorturl.at/aejkF) tests, if necessary~
- [x] I have updated relevant documentation, if necessary
- [x] I have performed a self-review of my code
- [x] I have manually tested this PR in the deployed cloud environment

gmrabian committed:
remove references to branch master

gmrabian commented on README.md:
```diff
@@ -115,10 +115,7 @@ On the SEDS side, this topic is updated on every submission of seds data, but CA
-
-- [generateEnrollmentTotals](https://github.com/Enterprise-CMCS/macpro-mdct-seds/blob/master/services/app-api/handlers/state-forms/post/generateEnrollmentTotals.js)
-- [updateStateForms](https://github.com/Enterprise-CMCS/macpro-mdct-seds/blob/master/services/app-api/handlers/state-forms/post/updateStateForms.js)
+Updates outside of that time frame will need to be manually corrected in CARTS, or the integration will need to be modifed to collect data for old forms. CARTS additionally looks for the `enrollmentCounts` property which is only included in forms 21E and 64.21E (question 7), either by manual trigger or update.
```
are there new files in `main` that we want to reference here?

ntsummers1 replied:
I don't think so.. @BearHanded? I don't think theres anything new related to this, right?

gmrabian commented:
> I think the `master` in `.github/workflows/destroy.yml` is referencing the master branch of `Enterprise-CMCS/bigmac`, and not of this repo - is that correct?

Correct. The [documentation](https://github.com/benc-uk/workflow-dispatch?tab=readme-ov-file#ref) states we need to specify ref to ensure we use the master branch of that action, rather than have GitHub infer our branch name for the same.

gmrabian commented:
> I don't have any problem with a branch named `skipci-archive-master`. I do think `skipci-archive-carts-v2` may be a better choice, because it's more direct; it doesn't require you to remember the factoid "we switched from master to main when we rewrote the app and called it v3". Although it does require you to remember the factoid "the current incarnation of CARTS is called v3" 🤔
> 
> As for how long we would keep it around, I think the answer is "forever", right? I mean, it's a large part of the history of the project, and that's a large part of what Git is for.

I asked for feedback on the branch name for the same reasons you mentioned. I think it might be good to keep `master` out of it if possible. Maybe `skipci-archive-carts-old`? Not a big deal regardless.

Good point about Git history
